hyperspy/signal.py

```diff
@@ -4994,7 +4994,7 @@ def _map_iterate(
             testing_kwargs = {}
             for ikey, key in enumerate(arg_keys):
                 test_ind = (0,) * len(os_am.navigation_axes)
-                testing_kwargs[key] = np.squeeze(args[ikey][test_ind]).compute()
+                testing_kwargs[key] = np.squeeze(args[ikey][test_ind].compute())[()]
```

Comment:
I think that this is good but we should think about adding in a couple more tests if that is that case. Currently if I remove the `np.squeeze`  and `[()]` all of the tests still pass. 

I'll do some more testing later today but it is possible that the squeeze function isn't necessary.  I think older code used `inav` instead of operating directly on the data which meant that it failed when there was ragged signal passed and `np.squeeze` wasn't called.